### PR TITLE
[Feature] Add flag to disable year picker toggle action

### PR DESCRIPTION
**Pull request by IronLad85**

### CODE REVIEW

File: lib/src/widgets/calendar_date_picker2.dart [hunk 1]
```diff
@@ -72,6 +74,9 @@ class CalendarDatePicker2 extends StatefulWidget {
   /// The calendar UI related configurations
   final CalendarDatePicker2Config config;
 
+  /// Flag to disable showing the year picker.
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2.dart:77]: Please update this name and description to `disableYearPicker` and `Flag to disable year picker and hide the toggle icon`

File: lib/src/widgets/calendar_date_picker2.dart [hunk 2]
```diff
@@ -35,6 +36,7 @@ class CalendarDatePicker2 extends StatefulWidget {
     required this.config,
     this.onValueChanged,
     this.onDisplayedMonthChanged,
+    this.disableYearPickerToggle = false,
```
[the remark this answers is not in the document]
Comment (IronLad85) [reply lib/src/widgets/calendar_date_picker2.dart]: Fixed it

File: lib/src/widgets/calendar_date_picker2.dart [hunk 3]
```diff
@@ -351,6 +361,10 @@ class _DatePickerModeToggleButtonState
     with SingleTickerProviderStateMixin {
   late AnimationController _controller;
 
+  bool get canTogglePicker {
```
[the remark this answers is not in the document]
Comment (IronLad85) [reply lib/src/widgets/calendar_date_picker2.dart]: Fixed it

File: lib/src/models/calendar_date_picker2_config.dart [hunk 4]
```diff
@@ -41,6 +41,7 @@ class CalendarDatePicker2Config {
     this.selectableDayPredicate,
     this.dayTextStylePredicate,
     this.dayBuilder,
+    this.disableYearPicker = false,
```
Comment (theideasaler) [inline lib/src/models/calendar_date_picker2_config.dart:44]: please leave the default value as null.

File: lib/src/models/calendar_date_picker2_config.dart [hunk 5]
```diff
@@ -124,6 +125,9 @@ class CalendarDatePicker2Config {
   /// Function to build customizable day widget
   final CalendarDayBuilder? dayBuilder;
 
+  /// Flag to disable year picker and hide the toggle icon.
```
Comment (theideasaler) [inline lib/src/models/calendar_date_picker2_config.dart:128]: could you please also add this config into `copyWith` and `CalendarDatePicker2WIthActionButtonsConfig` and its `copyWith` method

File: lib/src/models/calendar_date_picker2_config.dart (hunk 4 above)
Comment (IronLad85) [reply lib/src/models/calendar_date_picker2_config.dart]: okay

File: lib/src/models/calendar_date_picker2_config.dart (hunk 5 above)
Comment (IronLad85) [reply lib/src/models/calendar_date_picker2_config.dart:128]: yep

**Review by theideasaler — APPROVED** [review]

**Comment by christianschoenmakers** [comment]
I want to disable the date picker in the widget. This PR seems to make this work, but can't find this setting in the current config. Is this replaced with something else?

**Comment by vt-erikchow** [comment]
@christianschoenmakers 
seems like it change it to `disableModePicker`

**Pull request merged**